Sources/CarPlay/OACarPlayDashboardInterfaceController.mm

```diff
@@ -572,24 +572,36 @@ - (CPManeuver *)createTurnManeuver:(CPTravelEstimates *)estimates directionInfo:
     [turnDrawable setNeedsDisplay];
 
     CPManeuver *maneuver = [[CPManeuver alloc] init];
-    maneuver.symbolImage = [turnDrawable toUIImage];
+    if (estimates)
+        // automatic trimming of the text from the end and automatic resizing of the icon to a minimum of 1px
+        maneuver.symbolImage = [turnDrawable toUIImage];
     maneuver.initialTravelEstimates = estimates;
     maneuver.userInfo = @{ @"imminent" : @(directionInfo.imminent) };
     NSString *streetName = [self defineStreetName:directionInfo];
-    
+
     if (streetName)
     {
-        if (estimates) {
+        if (estimates)
+        {
             maneuver.instructionVariants = @[streetName];
         }
         else
         {
-            NSString *distanceString = [OAOsmAndFormatter getFormattedDistance:directionInfo.distanceTo withParams:[OsmAndFormatterParams useLowerBounds]];
-            NSString *instruction = [NSString stringWithFormat:@"%@, %@", distanceString, streetName];
-            maneuver.instructionVariants = @[instruction];
+            // shows full text without trimming and fixes the icon size
+            NSString *distanceString = [OAOsmAndFormatter getFormattedDistance:directionInfo.distanceTo
+                                                                    withParams:[OsmAndFormatterParams useLowerBounds]];
+            NSTextAttachment *attachment = [[NSTextAttachment alloc] init];
+            attachment.image = [OAUtilities resizeImage:[turnDrawable toUIImage] newSize:CGSizeMake(16, 16)];
```

Comment:
Why not set correct `turnDrawable.frame` instead?

Reply:
because this method makes the wrong size and then scales to artifacts